components/rich_text_editor/extensions/emoji/suggestion.js

```diff
@@ -0,0 +1,106 @@
+import { VueRenderer } from '@tiptap/vue-3';
+import { emojisIndexed } from '@/components/emoji_picker/emojis';
+
+import EmojiList from './EmojiList.vue';
+import { createTippy } from '@/components/popover/tippy_utils';
+
+export default {
+  items: ({ query }) => {
+    const emojiList = Object.values(emojisIndexed);
+    const filteredEmoji = emojiList.filter(function (item) {
+      if (this.count < 5 &&
+        item.shortname.substring(1, item.shortname.length - 1).startsWith(query.toLowerCase())) {
+        this.count++;
+        return true;
+      }
+      return false;
+    }, { count: 0 });
+    return filteredEmoji.map(item => item.shortname);
+  },
+
+  command: ({ editor, range, props }) => {
+    // increase range.to by one when the next node is of type "text"
+    // and starts with a space character
+    const nodeAfter = editor.view.state.selection.$to.nodeAfter;
+    const overrideSpace = nodeAfter?.text?.startsWith(' ');
+
+    if (overrideSpace) {
+      range.to += 1;
+    }
+
+    editor
+      .chain()
+      .focus()
+      .insertContentAt(range, [
+        {
+          type: 'emoji',
+          attrs: props,
+        },
+        {
+          type: 'text',
+          text: ' ',
+        },
+      ])
+      .run();
+
+    window.getSelection()?.collapseToEnd();
+  },
+
+  render: () => {
+    let component;
+    let popup;
+
+    return {
+      onStart: props => {
+        component = new VueRenderer(EmojiList, {
+          props,
+          editor: props.editor,
+        });
+
+        if (!props.clientRect) {
+          return;
+        }
+
+        const tippyOptions = {
+          getReferenceClientRect: props.clientRect,
+          appendTo: () => document.body,
+          content: component.element,
+          showOnCreate: true,
+          interactive: true,
+          trigger: 'manual',
+          placement: 'bottom-start',
```

Comment:
any fallbackPlacement needed in case there isn't enough room to display?

Reply:
the tippy props do not have fallback placement. in the docs i see popper options having that option to be set. Should i do that way?

Reply:
Since it is working and this is a side effect, it is okay to go with this at this point.

Would be cool to try to trigger how it is when there is no space to display and see whats going on and validate if it is necessary to do sth, because at this point we dont know.

If it is necessary you can create a ticket to fix it later.

Reply:
https://dialpad.atlassian.net/browse/DLT-1202